### ISSUE TITLE
CI Failures

### ISSUE THREAD

**Issue by SergioRAgostinho** [PointCloudLibrary/pcl#3288]
Couple of observations:
* All osx jobs starting from August have failed in the same way:
  * at approximately the same time (40 or 41 minute into the build)
  * at the same percentage (35%) after completing `pcl_surface` target
* Before August osx jobs occasionally failed, but with "communication lost" error
* Before August in succeeding jobs `pcl_surface` target was at 36% of the build

I also checked updates that happened to Azure osx image recently ([week 29](https://github.com/microsoft/azure-pipelines-image-generation/pull/1093), [week 30](https://github.com/microsoft/azure-pipelines-image-generation/pull/1147)), but there is nothing significant that can affect us. The only somewhat remotely related change is a tiny bump in homebrew version.

_Originally posted by @taketwo in https://github.com/PointCloudLibrary/pcl/pull/3270#issuecomment-520155410_

**Comment by SergioRAgostinho** [comment]
> Our Ubuntu job has been failing lately as well. Coincidentally during compilation of examples for the surface module:
> 
> ```
> [ 41%] Building CXX object examples/surface/CMakeFiles/pcl_example_nurbs_fitting_closed_curve.dir/example_nurbs_fitting_closed_curve.cpp.o
> /__w/1/s/examples/surface/example_nurbs_fitting_closed_curve.cpp:1:55: fatal error: pcl/surface/on_nurbs/fitting_curve_2d_pdm.h: No such file or directory
> compilation terminated.
> examples/surface/CMakeFiles/pcl_example_nurbs_fitting_closed_curve.dir/build.make:62: recipe for target 'examples/surface/CMakeFiles/pcl_example_nurbs_fitting_closed_curve.dir/example_nurbs_fitting_closed_curve.cpp.o' failed
> make[2]: *** [examples/surface/CMakeFiles/pcl_example_nurbs_fitting_closed_curve.dir/example_nurbs_fitting_closed_curve.cpp.o] Error 1
> make[1]: *** [examples/surface/CMakeFiles/pcl_example_nurbs_fitting_closed_curve.dir/all] Error 2
> ```
> 
> I can not reproduce on my 16.04 machine.


_Originally posted by @taketwo in https://github.com/PointCloudLibrary/pcl/pull/3270#issuecomment-520159574




> > I can not reproduce on my 16.04 machine.
> 
> Neither on 18.04 with clang (as usual: full build)


_Originally posted by @SunBlack  in https://github.com/PointCloudLibrary/pcl/pull/3270#issuecomment-520252830

**Comment by SergioRAgostinho** [comment]
I've just migrate the CI issues thread here, so we can let the original one die.  I've just rebuilt the lib in osx with the latest changes, with surface on nurbs included. Besides the multiple warnings popping, everything is building. 

For the ubuntu build, since it is a docker image, it is worth to see if such behavior can be replicated from within that environment. 

For the osx build, I need to investigate further.

**Comment by taketwo** [comment]
> For the ubuntu build, since it is a docker image, it is worth to see if such behavior can be replicated from within that environment.

Good point! Unfortunately, I can not replicate it in Docker either.

I have noticed a curious thing though. On Azure we get the [following output](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=6049) in the configuration summary:

```
-- The following subsystems will not be built:
--   examples: Requires surface.
```

Despite this, examples are actually built (and lead to errors). When I try to reproduce this inside Docker on my workstation this configuration output snippet is not present. So far I have no idea what causes the difference and why on Azure our buildsystem thinks that examples are not built. Will need to investigate this further.

**Comment by taketwo** [comment]
Current status: OSX jobs are still failing, Ubuntu jobs are fixed.

**Comment by SunBlack** [comment]
What was the issue?

**Comment by taketwo** [comment]
With Ubuntu? `sufrace` module missing in the dependencies of `examples` target. See #3295.

**Comment by SunBlack** [comment]
> * All osx jobs starting from August have failed in the same way:
>   
>   * at the same percentage (35%) after completing `pcl_surface` target

It's failing a lot of earlier:
```
make[2]: *** No rule to make target `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/OpenGL.framework', needed by `lib/libpcl_simulation.1.9.1.99.dylib'.  Stop.
make[2]: *** Waiting for unfinished jobs....
```

**Comment by taketwo** [comment]
Where? As far as I can see, it is still failing at 35% in the very last CI run: https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=6313

**Comment by SunBlack** [comment]
This message already occurs at 24% (line 4722 in your references log). When you change `cmake --build . -- -j2` to `cmake --build . -- -j1` we would see error message at end (I believe).

**Comment by taketwo** [comment]
Oh, you are right, we overlooked that! So this error has something to do with OpenGL.

**Comment by kunaltyagi** [comment]
Apparently, XCode 10 has some issues with OpenGL

**Comment by taketwo** [comment]
It's annoying to see one of the CI jobs red all the time. So yes, I also think that disabling `simulation` for the OSX job may be a good idea for the time being.

**Comment by taketwo** [comment]
Closed by #3358.
